src/foam/dao/RemoveSink.java

```diff
@@ -6,25 +6,32 @@
 import foam.mlang.order.Comparator;
 import foam.mlang.predicate.Predicate;
 
-public class RemoveSink {
-
-  public RemoveSink(final DAO dao) {
-    final X x = ((AbstractDAO) dao).getX();
-    dao.select_(x, new AbstractSink() {
-      @Override
-      public void put(FObject obj, Detachable sub) {
-        dao.remove_(x, obj);
-      }
-    }, 0, AbstractDAO.MAX_SAFE_INTEGER,null, null);
+public class RemoveSink implements Sink {
+
+  protected DAO dao;
+
+  public RemoveSink(DAO dao) {
+    this.dao = dao;
+  }
+
+  @Override
+  public void put(FObject obj, Detachable sub) {
+    final X x = ((AbstractDAO) this.dao).getX();
```

Comment:
Replace these two lines with just: dao.remove(obj);

dao already has a context and you don't need "this." in Java.